F811 Redefinition of unused import not silenced by noqa for grouped imports

When there are conditional (re)imports which are subsequently exported the `noqa` directive doesn't silence the F811 rule. 

This can be demonstrated by the following example:
- reimport.py
  ```python
  from typing import (
      TYPE_CHECKING,
      Dict,
      List,
  )

  if not TYPE_CHECKING:
      from beartype.typing import (  # noqa: F811
          Dict,
          List
      )

  __all__ = [
      "Dict",
      "List"
  ]
  ```
  
  > **Note**
  > If I change the second import to:
  > ```
  > if not TYPE_CHECKING:
  >    from beartype.typing import Dict  # noqa: F811
  >    from beartype.typing import List  # noqa: F811
  > ```
  > the `noqa` is respected.

- ruff version:
  ```shell
  $ ruff --version
  ruff 0.0.225
  ```
- ruff output:
  ```shell
  $ ruff reimport.py
  reimport.py:11:9: F811 Redefinition of unused `Dict` from line 5
  reimport.py:12:9: F811 Redefinition of unused `List` from line 6
  Found 2 error(s).
  ```

--- a/resources/test/fixtures/pyflakes/F811_21.py
+++ b/resources/test/fixtures/pyflakes/F811_21.py
@@ -1,0 +1,53 @@
+"""Test: noqa directives."""
+
+from typing_extensions import List, Sequence
+
+# This should ignore both errors.
+from typing import (  # noqa: F811
+    List,
+    Sequence,
+)
+
+# This should ignore both errors.
+from typing import (  # noqa
+    List,
+    Sequence,
+)
+
+# This should ignore both errors.
+from typing import (
+    List,  # noqa: F811
+    Sequence,  # noqa: F811
+)
+
+# This should ignore both errors.
+from typing import (
+    List,  # noqa
+    Sequence,  # noqa
+)
+
+# This should ignore the first error.
+from typing import (
+    List,  # noqa: F811
+    Sequence,
+)
+
+# This should ignore both errors.
+from typing import (  # noqa
+    List,
+    Sequence,
+)
+
+# This should ignore both errors.
+from typing import List, Sequence  # noqa: F811
+
+# This should ignore both errors.
+from typing import List, Sequence  # noqa
+
+
+def f():
+    # This should ignore both errors.
+    from typing import (  # noqa: F811
+        List,
+        Sequence,
+    )

--- a/src/checkers/ast.rs
+++ b/src/checkers/ast.rs
@@ -3858,13 +3858,21 @@ impl<'a> Checker<'a> {
                             ))
                     {
                         if self.settings.rules.enabled(&Rule::RedefinedWhileUnused) {
-                            self.diagnostics.push(Diagnostic::new(
+                            let mut diagnostic = Diagnostic::new(
                                 pyflakes::rules::RedefinedWhileUnused {
                                     name: name.to_string(),
                                     line: existing.range.location.row(),
                                 },
                                 binding_range(&binding, self.locator),
-                            ));
+                            );
+                            if let Some(parent) = binding.source.as_ref() {
+                                if matches!(parent.node, StmtKind::ImportFrom { .. })
+                                    && parent.location.row() != binding.range.location.row()
+                                {
+                                    diagnostic.parent(parent.location);
+                                }
+                            }
+                            self.diagnostics.push(diagnostic);
                         }
                     }
                 } else if existing_is_import && binding.redefines(existing) {
@@ -4614,13 +4622,22 @@ impl<'a> Checker<'a> {
 
                         if let Some(indices) = self.redefinitions.get(index) {
                             for index in indices {
-                                diagnostics.push(Diagnostic::new(
+                                let rebound = &self.bindings[*index];
+                                let mut diagnostic = Diagnostic::new(
                                     pyflakes::rules::RedefinedWhileUnused {
                                         name: (*name).to_string(),
                                         line: binding.range.location.row(),
                                     },
-                                    binding_range(&self.bindings[*index], self.locator),
-                                ));
+                                    binding_range(rebound, self.locator),
+                                );
+                                if let Some(parent) = &rebound.source {
+                                    if matches!(parent.node, StmtKind::ImportFrom { .. })
+                                        && parent.location.row() != rebound.range.location.row()
+                                    {
+                                        diagnostic.parent(parent.location);
+                                    }
+                                };
+                                diagnostics.push(diagnostic);
                             }
                         }
                     }

--- a/src/rules/pyflakes/mod.rs
+++ b/src/rules/pyflakes/mod.rs
@@ -89,6 +89,7 @@ mod tests {
     #[test_case(Rule::RedefinedWhileUnused, Path::new("F811_18.py"); "F811_18")]
     #[test_case(Rule::RedefinedWhileUnused, Path::new("F811_19.py"); "F811_19")]
     #[test_case(Rule::RedefinedWhileUnused, Path::new("F811_20.py"); "F811_20")]
+    #[test_case(Rule::RedefinedWhileUnused, Path::new("F811_21.py"); "F811_21")]
     #[test_case(Rule::UndefinedName, Path::new("F821_0.py"); "F821_0")]
     #[test_case(Rule::UndefinedName, Path::new("F821_1.py"); "F821_1")]
     #[test_case(Rule::UndefinedName, Path::new("F821_2.py"); "F821_2")]

